src/test/storage/dictionary_segment_test.cpp

```diff
@@ -139,4 +139,18 @@ TEST_F(StorageDictionarySegmentTest, MemoryUsageUInt32) {
             ((UINT16_MAX + 2)) * sizeof(int32_t) + ((UINT16_MAX + 2)) * sizeof(uint32_t));
 }
 
+TEST_F(StorageDictionarySegmentTest, AccessElementsViaGet) {
+  value_segment_int->append(1);
+  value_segment_int->append(2);
+
+  std::shared_ptr<AbstractSegment> segment;
+  segment = std::make_shared<DictionarySegment<int32_t>>(value_segment_int);
+  auto dict_segment = std::dynamic_pointer_cast<DictionarySegment<int32_t>>(segment);
```

Comment:
This can be simplified to
```suggestion
  auto dict_segment = std::make_shared<DictionarySegment<int32_t>>(value_segment_int);
  auto segment = static_cast<std::shared_ptr<AbstractSegment>>(dict_segment);
```